src/setup_payload/tests/TestManualCode.cpp

```diff
@@ -153,6 +153,40 @@ void assertPayloadValues(nlTestSuite * inSuite, CHIP_ERROR actualError, CHIP_ERR
     NL_TEST_ASSERT(inSuite, payload.productID == productID);
 }
 
+void TestGenerateAndParser_ManualSetupCodeWithLongDiscriminator(nlTestSuite * inSuite, void * inContext)
+{
+    SetupPayload payload       = GetDefaultPayload();
+    payload.requiresCustomFlow = false;
+    payload.discriminator      = 0xf11;
+
+    {
+        // Test short 11 digit code
+        ManualSetupPayloadGenerator generator(payload);
+        string result;
+        NL_TEST_ASSERT(inSuite, generator.payloadDecimalStringRepresentation(result) == CHIP_NO_ERROR);
+
+        SetupPayload outPayload;
+        CHIP_ERROR err = ManualSetupPayloadParser(result).populatePayload(outPayload);
+        assertPayloadValues(inSuite, err, CHIP_NO_ERROR, outPayload, payload.setUpPINCode, 1, payload.vendorID, payload.productID);
```

Comment:
I wonder whether it would be clearer to have the discriminator be `0xf1a` or something and pass `0xa` here as the 6th argument, to make it clearer how the two are connected.  And maybe `0xf1b` for the second test?

Reply:
Sure, I'll update the test.